packages/application-components/src/components/tab-header/tab-header.spec.tsx

```diff
@@ -0,0 +1,51 @@
+import { warning } from '@commercetools-uikit/utils';
+import { screen, renderComponent, fireEvent, waitFor } from '../../test-utils';
+import TabHeader from './tab-header';
+
+jest.mock('@commercetools-uikit/utils', () => ({
+  ...jest.requireActual('@commercetools-uikit/utils'),
+  warning: jest.fn(),
+}));
+
+const renderTabHeader = (additionalProps = {}) =>
+  renderComponent(
+    <TabHeader to="/tab-one" label="Tab One" {...additionalProps} />
+  );
+
+describe('rendering', () => {
+  it('should render when "to" is string', () => {
+    renderTabHeader();
+
+    screen.getByText(/tab one/i);
+  });
+  it('should render when "to" is of LocationProvider type', () => {
+    renderTabHeader({ to: { pathname: '/tab-one' } });
+
+    screen.getByText(/tab one/i);
+  });
+  it('should display a custom intl message instead of a label', () => {
+    const intlMessage = { id: 'localized', defaultMessage: 'localized text' };
+    renderTabHeader({ intlMessage });
+    screen.getByText('localized text');
+  });
+});
+describe('navigation', () => {
+  it('should navigate to link when clicked', async () => {
+    const { history } = renderTabHeader();
+
+    fireEvent.click(screen.getByRole('link', { name: /tab one/i }));
```

Comment:
Maybe use `tab` as the role?

Reply:
Unfortunately `fireEvent.click(screen.getByRole('tab'));` doesn't seem to work currently (doesn't trigger the underlaying `<Link>`'s action). 
We would need to set the `<Link>`'s role to `tab` to make it work.
Are you OK with that change?